mapper.go

```diff
@@ -0,0 +1,308 @@
+	return NewFuncMapper(func(src T) (any, error) {
+		for _, mp := range mappers {
+			if result, err := mp.Map(src); err == nil {
+				return result, nil
+			}
+		}
+		return nil, NewErrUnexpectedValue(src)
+	})
+}
+
+// replaceSeparators replaces ignore characters and decimal separators.
+func replaceSeparators(src, ignoreChars, decSeparators string) string {
+	for _, char := range ignoreChars {
+		src = strings.ReplaceAll(src, string(char), "")
+	}
+	for _, char := range decSeparators {
+		src = strings.ReplaceAll(src, string(char), ".")
+	}
+	return src
+}
+
+// BoolParser is a parser to bool.
+type BoolParser struct{}
+
+// NewBoolParser creates BoolParser.
+func NewBoolParser() *BoolParser {
+	return &BoolParser{}
+}
+
+// Map is the implementation of Mapper[string] for BoolParser.
+func (*BoolParser) Map(src string) (any, error) {
+	return strconv.ParseBool(src)
+}
+
+// UIntParser is a parser to uint64.
+type UIntParser struct {
+	ignoreChars string
+}
+
+// NewUIntParser creates UIntParser.
+func NewUIntParser(ignoreChars string) *UIntParser {
+	return &UIntParser{ignoreChars: ignoreChars}
+}
+
+// Map is the implementation of Mapper[string] for UIntParser.
+func (parser *UIntParser) Map(src string) (any, error) {
+	src = replaceSeparators(src, parser.ignoreChars, "")
+	return strconv.ParseUint(src, 10, 64)
+}
+
+// IntParser is a parser to int64.
+type IntParser struct {
+	ignoreChars string
+}
+
+// NewIntParser creates IntParser.
+func NewIntParser(ignoreChars string) *IntParser {
+	return &IntParser{ignoreChars: ignoreChars}
+}
+
+// Map is the implementation of Mapper[string] for IntParser.
+func (parser *IntParser) Map(src string) (any, error) {
+	src = replaceSeparators(src, parser.ignoreChars, "")
+	return strconv.ParseInt(src, 10, 64)
+}
+
+// FloatParser is a parser to float64.
+type FloatParser struct {
+	ignoreChars   string
+	decSeparators string
+}
+
+// NewFloatParser creates FloatParser.
+func NewFloatParser(ignoreChars, decSeparators string) *FloatParser {
+	return &FloatParser{ignoreChars: ignoreChars, decSeparators: decSeparators}
+}
+
+// Map is the implementation of Mapper[string] for FloatParser.
+func (parser *FloatParser) Map(src string) (any, error) {
+	src = replaceSeparators(src, parser.ignoreChars, parser.decSeparators)
+	return strconv.ParseFloat(src, 64)
+}
+
+// DecimalParser is a parser to decimal.Decimal.
+type DecimalParser struct {
+	ignoreChars   string
+	decSeparators string
+}
+
+// NewDecimalParser creates DecimalParser.
+func NewDecimalParser(ignoreChars, decSeparators string) *DecimalParser {
+	return &DecimalParser{ignoreChars: ignoreChars, decSeparators: decSeparators}
+}
+
+// Map is the implementation of Mapper[string] for DecimalParser.
+func (parser *DecimalParser) Map(src string) (any, error) {
+	src = replaceSeparators(src, parser.ignoreChars, parser.decSeparators)
+	result, err := decimal.NewDecimalFromString(src)
+	if err == nil {
+		return *result, nil
+	}
+	return nil, err
```

Comment:
A usual way to handle an error:
```suggestion
	if err != nil {
		return nil, fmt.Errorf("unexpected decimal: %w", err)
	}
	return *result, nil
```

Reply:
Fixed.